ts/src/transports/Transport.ts

```diff
@@ -59,3 +60,7 @@ function detectTransport(): TransportConstructor {
 
   throw new Error("No suitable transport found for gRPC-Web");
```

Comment:
Any chance we could add the WebsocketTransport to auto-detected transports?

Reply:
I'd be keen to separate the implementation of the transport (this PR) from a discussion around how this should be used.

Currently the default transport factory just chooses the transport through a fallback process, but Websockets aren't necessarily the first in that list of transports to fallback through as they have some potentially undesirable tradeoffs.

I'd be happy to look at a PR that proposes a good standard way of using this transport after this lands 👍